python_modules/libraries/dagster-dask/dagster_dask/config.py

```diff
@@ -0,0 +1,154 @@
+                ),
+                "asynchronous": Field(
+                    Bool,
+                    description="Set to True if using this client within async/await functions.",
+                    is_required=False,
+                ),
+                "name": Field(
+                    String,
+                    description="Gives the client a name that will be included in logs generated on the scheduler.",
+                    is_required=False,
+                ),
+                "direct_to_workers": Field(
+                    Bool,
+                    description="Whether or not to connect directly to the workers.",
+                    is_required=False,
+                ),
+                "heartbeat_interval": Field(
+                    Int,
+                    description="Time in milliseconds between heartbeats to scheduler.",
+                    is_required=False,
+                ),
+            }
+        ),
+        description="Dask distributed client options.",
+        is_required=False,
+    ),
+    # The cluster config fields consist of a cluster type keys and permissive dicts of key-values
+    # that will be passed in to the corresponding cluster class. Only a single cluster type may be
+    # specified at a time.
+    "cluster": Field(
+        Selector(
+            {
+                **{
+                    key: Field(
+                        Permissive(),
+                        is_required=False,
+                        description=f"{meta['name']} cluster config. Requires {meta['module']}.",
+                    )
+                    for key, meta in DaskClusterTypes.items()
+                },
+                **{
+                    "existing": Field(
+                        {"address": StringSource},
+                        description="Connect to an existing scheduler (deprecated, use client address config).",
+                    )
+                },
+            }
+        ),
+        description="Create a Dask cluster. Will be passed as the client address.",
+        is_required=False,
+    ),
+}
+
+
+def create_from_config(config):
+    """Create Dask Client and Cluster objects based on the provided config, which may be
+    an executor_config or resource_config.
+    """
+
+    cluster_config = check.opt_nullable_dict_param(config.get("cluster"), "cluster")
+    client_config = check.opt_dict_param(config.get("client"), "client")
+
+    # Construct a cluster object.
+    if cluster_config:
+        # Unpack the cluster type and init options. The cluster_config should consist
+        # of a single dict member because the `cluster` field is defined as a selector.
+        cluster_type, cluster_opts = next(iter(cluster_config.items()))
+
+        # Deprecated option for specifying an existing cluster by its scheduler address.
+        if cluster_type == "existing":
+            cluster = None
+            client_config["address"] = cluster_opts["address"]
```

Comment:
can we make sure both ways are under test? I think only the deprecated version is being tested since you haven't added any

Reply:
Good point. I'll add a test for that.

Reply:
Added a test for `client:address:`. Also, added a warning if `cluster:existing:address:` is used.